ggshield/utils/archive.py

```diff
@@ -0,0 +1,124 @@
+import shutil
+import stat
+import tarfile
+from pathlib import Path
+from typing import List, Optional, Tuple
+from zipfile import ZipFile, ZipInfo
+
+
+class UnsafeArchive(Exception):
+    """
+    Raised when an archive is considered unsafe
+    """
+
+    def __init__(self):
+        self.bad_paths: List[Path] = []
+        self.bad_links: List[Tuple[Path, Path]] = []
+
+    def __str__(self) -> str:
+        lines: List[str] = ["Unsafe archive."]
```

Comment:
I think it could be good to be a bit more explicit ?
```suggestion
        lines: List[str] = ["Unsafe archive, it could cause to write/overwrite or link outside the target folder when unpacking. To avoid vulnerabilities to malicious attack ggshield refuses to unpack it."]
```